lib/jekyll/filters/url_filters.rb

```diff
@@ -29,6 +29,23 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces the path to the given asset relative to the assets directory at the
+      # domain root based on site.baseurl
+      #
+      # input - the asset file relative to the 'assets' directory. The string may also
+      #         include any subfolder(s) the asset may be a part of.
+      #           e.g. "main.css" or "css/main.css"
+      #
+      # Returns a path relative to the assets directory as a String.
+      def asset_url(input)
+        return if input.nil?
+        site = @context.registers[:site]
+        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
```

Comment:
You are so close to the edge of the method it should just be:

```
if site.config["baseurl"].nil?
  # This
else
  # That
end
```

Explicit returns have a cost in Ruby that they shouldn't (but they do) and the `if` + `return` will result in a higher cost than an `if/else` in this case... especially when at the edge of the method.